src/Raven.Server/Documents/Indexes/Static/TimeSeries/MapTimeSeriesIndex.cs

```diff
@@ -0,0 +1,213 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Raven.Client;
+using Raven.Client.Documents.Changes;
+using Raven.Client.Documents.Indexes;
+using Raven.Client.Documents.Indexes.TimeSeries;
+using Raven.Server.Config;
+using Raven.Server.Documents.Indexes.Configuration;
+using Raven.Server.Documents.Indexes.Persistence.Lucene;
+using Raven.Server.Documents.Indexes.Workers;
+using Raven.Server.Documents.Indexes.Workers.TimeSeries;
+using Raven.Server.Documents.Queries;
+using Raven.Server.ServerWide.Context;
+
+namespace Raven.Server.Documents.Indexes.Static.TimeSeries
+{
+    public class MapTimeSeriesIndex : MapIndexBase<MapIndexDefinition, IndexField>
+    {
+        private readonly HashSet<string> _referencedCollections = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+
+        protected internal readonly StaticTimeSeriesIndexBase _compiled;
+
+        private HandleReferences _handleReferences;
+
+        protected MapTimeSeriesIndex(MapIndexDefinition definition, StaticTimeSeriesIndexBase compiled)
+            : base(definition.IndexDefinition.Type, definition)
+        {
+            _compiled = compiled;
+
+            if (_compiled.ReferencedCollections == null)
+                return;
+
+            foreach (var collection in _compiled.ReferencedCollections)
+            {
+                foreach (var referencedCollection in collection.Value)
+                    _referencedCollections.Add(referencedCollection.Name);
+            }
+        }
+
+        protected override void SubscribeToChanges(DocumentDatabase documentDatabase)
+        {
+            if (documentDatabase == null)
+                return;
+
+            if (_referencedCollections.Count > 0)
+                documentDatabase.Changes.OnDocumentChange += HandleDocumentChange;
+
+            documentDatabase.Changes.OnTimeSeriesChange += HandleTimeSeriesChange;
+        }
+
+        protected override void UnsubscribeFromChanges(DocumentDatabase documentDatabase)
+        {
+            if (documentDatabase == null)
+                return;
+
+            if (_referencedCollections.Count > 0)
+                documentDatabase.Changes.OnDocumentChange -= HandleDocumentChange;
+
+            documentDatabase.Changes.OnTimeSeriesChange -= HandleTimeSeriesChange;
+        }
+
+        protected override void HandleDocumentChange(DocumentChange change)
+        {
+            if (_referencedCollections.Contains(change.CollectionName) == false)
+                return;
+
+            _mre.Set();
+        }
+
+        protected override IIndexingWork[] CreateIndexWorkExecutors()
+        {
+            var workers = new List<IIndexingWork>
+            {
+                //new CleanupDocuments(this, DocumentDatabase.DocumentsStorage, _indexStorage, Configuration, null)
```

Comment:
TODO:

we need this, because when document is delete, all timeseries are deleted and no tombstones are created.